R/recode-with-table.R

```diff
@@ -0,0 +1,1019 @@
+        conflict_vars <- intersect(feeder_vars, var_stack)
+        stop(
+          paste(
+            conflict_vars,
+            "is required to create",
+            variable_being_processed,
+            "but",
+            variable_being_processed,
+            "is needed to create",
+            conflict_vars
+          )
+        )
+      }
+
+      # Update var_stack and recurse to get the feeder vars
+      for (one_feeder in feeder_vars) {
+        # Need to check recoded data again in case a recursion added it
+        if (!one_feeder %in% names(recoded_data)) {
+          derived_return <-
+            recode_derived_variables(
+              variable_being_processed = one_feeder,
+              recoded_data = recoded_data,
+              variables_to_process = variables_to_process,
+              log = log,
+              print_note = print_note,
+              else_default = else_default,
+              label_list = label_list,
+              var_stack = var_stack
+            )
+          var_stack <- derived_return$var_stack
+          label_list <- derived_return$label_list
+          recoded_data <- derived_return$recoded_data
+          variables_to_process <-
+            derived_return$variables_to_process
+        }
+      }
+
+      # Obtain the function for each row
+      append(label_list, create_label_list_element(variable_rows))
+
+      row_being_checked <- variable_rows[row_num,]
+      func_cell <-
+        as.character(row_being_checked[[pkg.env$columns.CatValue]])
+      function_being_used <-
+        as.list(strsplit(func_cell, "::"))[[1]][[2]]
+
+      column_value <-
+        recoded_data %>%
+        dplyr::rowwise() %>%
+        dplyr::select(used_feeder_vars) %>%
+        dplyr::do(
+          column_being_added = calculate_custom_function_row_value(
+            .,
+            variable_names = used_feeder_vars,
+            custom_function_name = function_being_used
+          )
+        )
+      # Set type of var
+      if (as.character(row_being_checked[[pkg.env$columns.ToType]]) !=
+          pkg.env$columns.value.CatType) {
+        column_value <-
+          as.numeric(unlist(column_value[["column_being_added"]]))
+      } else{
+        column_value <-
+          as.factor(unlist(column_value[["column_being_added"]]))
+      }
+      recoded_data[[variable_being_processed]] <-
+        column_value
+
+      var_stack <-
+        var_stack[!(var_stack == variable_being_processed)]
+    }
+
+    return(
+      list(
+        var_stack = var_stack,
+        label_list = label_list,
+        recoded_data = recoded_data,
+        variables_to_process = variables_to_process
+      )
+    )
+  }
+calculate_custom_function_row_value <-
+  function(row_values,
+           variable_names,
+           custom_function_name) {
+    row_values <- unname(row_values)
+    custom_function_return_value <-
+      do.call(get(custom_function_name), row_values)
+
+    return(custom_function_return_value)
+  }
```

Comment:
```suggestion
}
```

Reply:
Due to the function being moved to the next line removing this space causes them to be out of line with each other